### PR TITLE
[TASK] Deprecate TCA subtypes

### PR DESCRIPTION
**Pull request by linawolf**
References: https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/1073
Releases: main

### CODE REVIEW

File: Documentation/ColumnsConfig/Type/Flex/Examples.rst [hunk 1]
```diff
@@ -32,39 +32,24 @@ Notice the :xml:`<input_1>` tag:
...
 
-The Data Structure for a FlexForm can also be loaded in the :sql:`pi_flexform`
+The data structure for a FlexForm can also be loaded in the :sql:`pi_flexform`
 field of the :sql:`tt_content` table by adding the following in the
-TCA Overrides of an extension:
+TCA Overrides of an extension, see this example from the EXT:blog_example:
 
-..  code-block:: php
-    :caption: EXT:my_extension/Configuration/TCA/Overrides/tt_content.php
+..  literalinclude:: _CodeSnippets/_tt_content_plugin.php
```
Comment (sfroemkenjw) [inline Documentation/ColumnsConfig/Type/Flex/Examples.rst:44]: Please activate linenumbers. IMO: lineos
Else it's hard to work with 18ff and 25ff below

File: Documentation/ColumnsConfig/Type/Flex/Examples.rst [hunk 2]
```diff
@@ -32,39 +32,24 @@ Notice the :xml:`<input_1>` tag:
...
+..  _columns-flex-example-plugin:
 
 FlexForm in a plugin
 ====================
 
-The Data Structure for a FlexForm can also be loaded in the :sql:`pi_flexform`
+The data structure for a FlexForm can also be loaded in the :sql:`pi_flexform`
 field of the :sql:`tt_content` table by adding the following in the
-TCA Overrides of an extension:
+TCA Overrides of an extension, see this example from the EXT:blog_example:
```
Comment (sfroemkenjw) [inline Documentation/ColumnsConfig/Type/Flex/Examples.rst:42]: I remember that we have a directive to link to extensions. Can we use it here?

File: Documentation/Types/SubtypeMigration.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+================================
+
+..  contents:: Table of contents
+
+..  _migration-subtype-flexform:
+
+Migrate plugins with FlexForms added via `subtypes_addlist`
+===========================================================
+
+If you used plugins with the now deprecated subtypes, you probably used
```
Comment (sfroemkenjw) [inline Documentation/Types/SubtypeMigration.rst:21]:
```suggestion
If you use plugins with the now deprecated subtypes, you probably used
```

File: Documentation/Types/SubtypeMigration.rst [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+:confval:`types-subtypes-addlist` to display a
+:ref:`FlexForm <t3coreapi:flexforms>` for configuration purposes.
+
+Migrate by adding the field :sql:`pi_flexform` with the utility method
+:php-short:`\TYPO3\CMS\Core\Utility\ExtensionManagementUtility` method
```
Comment (sfroemkenjw) [inline Documentation/Types/SubtypeMigration.rst:26]: Duplicate word "method"

File: Documentation/Types/SubtypeMigration.rst [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+`addToAllTCAtypes()` instead. You also have to change the parameters used for
+method `addPiFlexFormValue()`:
+
+..  literalinclude:: _CodeSnippets/_subtype_plugin_migration.diff
+    :caption: EXT:my_extension/Configuration/Overrides/tt_content.php
+
+The fields :sql:`pages` and :sql:`recursive` used to be added
+automatically to plugins when using the now outdated subtype "list-type".
```
Comment (sfroemkenjw) [inline Documentation/Types/SubtypeMigration.rst:34]: `list_type` with underscore?

```suggestion
automatically to plugins when using the now outdated subtype "list_type".
```

File: Documentation/Types/SubtypeMigration.rst [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+The same effect can now be used by simply not adding the fields in the first
+place:
+
+..  literalinclude:: _CodeSnippets/_subtype_plugin_exclude_migration.diff
+    :caption: EXT:my_extension/Configuration/Overrides/tt_content.php
+
+If any other fields have been removed with this method you can only remove
+them by overriding
+:confval:`$GLOBALS['TCA']['tt_content']['types'][$pluginSignature]['showitems] <types-showitem>`
```
Comment (sfroemkenjw) [inline Documentation/Types/SubtypeMigration.rst:53]:
```suggestion
:confval:`$GLOBALS['TCA']['tt_content']['types'][$pluginSignature]['showitem'] <types-showitem>`
```

File: Documentation/ColumnsConfig/Type/Flex/Examples.rst (hunk 1 above)
Comment (linawolf) [inline Documentation/ColumnsConfig/Type/Flex/Examples.rst:44]:
```suggestion
..  literalinclude:: _CodeSnippets/_tt_content_plugin.php
    :linenos:
```

File: Documentation/ColumnsConfig/Type/Flex/Examples.rst [hunk 7]
```diff
@@ -32,39 +32,25 @@ Notice the :xml:`<input_1>` tag:
...
```
Comment (linawolf) [inline Documentation/ColumnsConfig/Type/Flex/Examples.rst:42]:
```suggestion
TCA Overrides of an extension, see this example from the extension :composer:`t3docs/blog-example`:
```

File: Documentation/Types/SubtypeMigration.rst (hunk 4 above)
Comment (linawolf) [inline Documentation/Types/SubtypeMigration.rst:26]:
```suggestion
:php-short:`\TYPO3\CMS\Core\Utility\ExtensionManagementUtility`
```

File: Documentation/ColumnsConfig/Type/Flex/Examples.rst (hunk 2 above)
Comment (linawolf) [reply Documentation/ColumnsConfig/Type/Flex/Examples.rst]: done

File: Documentation/Types/SubtypeMigration.rst (hunk 4 above)
Comment (linawolf) [reply Documentation/Types/SubtypeMigration.rst]: done

**Review by sfroemkenjw — APPROVED** [review]

**Pull request merged**